Akka.Serialization.MessagePack/Formatters/SurrogatedFormatter.cs

```diff
@@ -0,0 +1,78 @@
+using System;
+using System.Collections.Concurrent;
+using Akka.Actor;
+using Akka.Util;
+using MessagePack;
+using MessagePack.Formatters;
+
+namespace Akka.Serialization.MessagePack.Resolvers
+{
+    class SurrogatedFormatterTools
+    {
+        internal static readonly ConcurrentDictionary<string, Type> deserTypeCache =
+            new ConcurrentDictionary<string, Type>();
+    }
+    public class SurrogatedFormatter<T> : IMessagePackFormatter<T>
```

Comment:
What is this for?

Reply:
If you mean `SurrogatedFormatter<T>`

We use this to convert `ISurrogated` Instances To/From messagepack format.
This lets us both do the proper `.FromSurrogate()` on deserialization, as well as let us handle surrogate types a bit more performantly than via the normal TypelessFormatter.

If you mean `SurrogatedFormatterTools`, we use it for caching the types of `ISurrogate` Instances, to again avoid TypelessFormatter.